### DISCUSSION TITLE
A simple periscope model does not behave as expected

### DISCUSSION THREAD

**Discussion by buddhasystem** [mitsuba-renderer/mitsuba3#1270] [Q&A] [answered]
Dear All,

I've built a simple periscope model, in which there is a spotlight on the left, shining horizontally, illuminating a disk tilted by 45 degree angle around the Y axis, thus reflecting the light downward. Below, there is a second disk also tilted by 45 degrees which reflects the light further such that the beam again is horizontal and hits the right wall, where I do expected to see a spot.

This only works if at least one of the disks is painted white, i.e. I see the faint spot right where I expect to see it, like in:
![image](https://github.com/user-attachments/assets/09ebdc61-fdf3-4b36-8245-5d7e10c72c84)

However, if both disks are mirror, I only see darkness. Since the walls are diffuse reflectors, I think it should work with two mirrors and it "kinda" works with the two bright white disks. But not with mirrors.

An advice will be appreciated. I can start tinkering with BSDF but really, a mirror illuminated by a conically shaped light source should just reflect it, correct?

**Comment by njroussel** [comment, marked as answer]
Hi again @buddhasystem  :smile: 

> a mirror illuminated by a conically shaped light source should just reflect it, correct?
Yes

As I pointed out in [this other discussion](https://github.com/mitsuba-renderer/mitsuba3/discussions/1243), creating valid light paths in the presence of perfect mirrors is (sometimes) impossible.

I'm assuming you're using a `path` integrator? If that's the case, the light paths are constructed from the sensor, to the light source. In this setup, it is impossible to build such a path. Reciprocally, the `ptracer` is a light-tracer, it starts tracing rays from the light source until it reaches the sensor. For this setup, the latter is the appropriate option.

**Reply by buddhasystem** [reply]
@njroussel Many thanks for the super relevant advice. With the _ptracer_ it actually works. It does take longer to render but that's 100% OK. Thank you! Below is the image of the periscope in operation.

![image](https://github.com/user-attachments/assets/7619bed1-21d5-4783-be14-7057c58c8695)
